app/src/main/java/com/anotherdeveloper/androidcourse/activities/A1.java

```diff
@@ -0,0 +1,167 @@
+import com.anotherdeveloper.androidcourse.utilities.ConnectivityManagement;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import io.realm.Realm;
+import io.realm.RealmResults;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class A1 extends AppCompatActivity implements
+        ConnectivityManagement.IConnectionCheck,
+        Realm.Transaction,
+        Realm.Transaction.OnSuccess,
+        Realm.Transaction.OnError,
+        Callback<Contact[]> {
+
+    ArrayList<Contact> contacts;
+    Realm realm;
+    ConnectivityManagement connectivityManagement;
+    boolean isConnected = false;
+
+    @BindView(R.id.contacts_recycler_view)
+    RecyclerView recyclerView;
+    @BindView(R.id.progress_bar)
+    ProgressBar progressBar;
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_a1);
+        setSupportActionBar(toolbar);
+        ButterKnife.bind(this);
+        contacts = new ArrayList<>();
+        initRecyclerView();
+        initConnectivityManagement();
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        realm.close();
+    }
+
+    @Override
+    public void onOnline(boolean isConnected) {
+
+            loadJSON();
+    }
+
+    @Override
+    public void onOffline(boolean isConnected) {
+
+            loadFromRealmDatabase();
+    }
+
+    @Override
+    public void execute(@NonNull Realm realm) {
+        realm.deleteAll();
+        for (Contact contact : contacts) {
+            realm.copyToRealmOrUpdate(contact);
+        }
+    }
+
+    @Override
+    public void onSuccess() {
+        Toast.makeText(getApplicationContext(), "Transacted to database", Toast.LENGTH_LONG).show();
+
+    }
+
+    @Override
+    public void onError(@NonNull Throwable error) {
+        Toast.makeText(getApplicationContext(), error.getMessage(), Toast.LENGTH_LONG).show();
+
+    }
+
+    @Override
+    public void onResponse(@NonNull Call<Contact[]> call, @NonNull Response<Contact[]> response) {
+        contacts.clear();
+        List<Contact> contactsFromAPI = Arrays.asList(response.body());
+        contacts = new ArrayList<>(contactsFromAPI);
+        refreshRecyclerView(R.layout.recycler_view_row);
+        saveIntoDatabase();
+        progressBar.setVisibility(View.GONE);
+    }
+
+    @Override
+    public void onFailure(@NonNull Call<Contact[]> call, @NonNull Throwable t) {
+        Toast.makeText(getApplicationContext(), t.getMessage(), Toast.LENGTH_SHORT).show();
+        progressBar.setVisibility(View.GONE);
+    }
+
+    private void initRecyclerView() {
+        recyclerView.setHasFixedSize(true);
+        recyclerView.setAdapter(new ContactsRecyclerViewAdapter(contacts, R.layout.recycler_view_row, this));
```

Comment:
aktywność nie musi wiedzieć, jaki jest id layoutu itemu listy, możesz to ustawić w Adapterze bezpośrednio i tylko tam tym zarządzać